internal/provider/string_data_source.go

```diff
@@ -0,0 +1,110 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &NameDataSource{}
+
+func NewNameDataSource() datasource.DataSource {
+	return &NameDataSource{}
+}
+
+// NameDataSource defines the data source implementation.
+type NameDataSource struct {
+	popts ProviderOpts
+}
+
+type NameDataSourceModel struct {
+	Id           types.String `tfsdk:"id"`
+	Name         types.String `tfsdk:"name"`
+	Registry     types.String `tfsdk:"registry"`
+	Repo         types.String `tfsdk:"repo"`
+	Digest       types.String `tfsdk:"digest"`
+	Tag          types.String `tfsdk:"tag"`
+	RegistryRepo types.String `tfsdk:"registry_repo"`
+}
+
+// Metadata implements datasource.DataSource.
+func (*NameDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_name"
+}
+
+// Schema implements datasource.DataSource.
+func (d *NameDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Reference string data source",
+
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				MarkdownDescription: "The name of an oci reference",
+				Required:            true,
+			},
+			"id": schema.StringAttribute{
+				MarkdownDescription: "The fully qualified oci reference",
+				Computed:            true,
+			},
+			"repo": schema.StringAttribute{
+				MarkdownDescription: "The repository of the oci reference",
+				Computed:            true,
+			},
+			"registry": schema.StringAttribute{
+				MarkdownDescription: "The registry of the oci reference",
+				Computed:            true,
+			},
+			"registry_repo": schema.StringAttribute{
+				MarkdownDescription: "The registry/repo of the oci reference",
```

Comment:
Maybe something more explicit like:

```suggestion
				MarkdownDescription: "Equivalent to '${registry}/${repo}'",
```